src/libs/collect.c

```diff
@@ -1723,9 +1723,13 @@ static void list_view(dt_lib_collect_rule_t *dr)
         {
           gchar *order_by = NULL;
           if(strcmp(dt_conf_get_string("plugins/collect/filmroll_sort"), "id") == 0)
```

Comment:
I see that some code was already there, but let's get this properly. The string returned by dt_conf_get_string must be freed.

So, please take the value before this line and free when not needed anymore.

But, note that I'm not sure this is useful. For example the "id descending" will always list the older collection first. Meaning that new collection will always be listed at the bottom of the list. I have more than 1000 collections and I don't see why I would hurt myself using this setting. And note also that this is a pref, meaning you set it once basically.

So all in all, I'm not ok to merge this as-is. Maybe others have different opinions, we'll see...

Reply:
@TurboGit : Thanks for your input, as I said: this is my first contribution, and I am happy about any help.

I don't want to have the most recent collections at the bottom of the list neither.
But unfortunately 'id' does not work for me, because I often import older shootings at a later point, so the list would be shuffled.
And the setting 'sort film rolls by folder' does show the most recent collections at the bottom, although I've also checked 'sort collection recent to older'.
That's my problem, and that's what by bug #6668 is all about.

The suggestion of this pull request is to add an option, which does exactly what I am missing: sort film-rolls by folder descending.

Just adding this one option would lead to the following list:
- id
- folder
- folder descending

As 'id' is also descending at the moment, this would be misleading in my opinion.
That's why I split it up:
- id
- id descending (default!)
- folder
- folder descending

You are absolutely right, that id (ascending) is quite useless, but in my opinion it makes the list more clear.
And of course id descending is set as default

But I am happy about concrete alternatives to solve my issue.

One more thing: I don't understand your point 'free the string'. I just added else clauses and did not change the logic. Could you tell me what exactly I have to change?